pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/DeleteAPIHybridClusterIntegrationTest.java

```diff
@@ -0,0 +1,247 @@
+    offlineSegmentsList.remove(0);
+    Assert.assertEquals(offlineSegmentsListReturn, offlineSegmentsList);
+  }
+
+  @Override
+  public void testGeneratedQueriesWithMultiValues() {}
+
+  @Test
+  public void deleteRealtimeSegmentFromGetAPI() throws Exception{
+    String segmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    JSONArray realtimeSegmentsList = getSegmentsFromJson(segmentList, "realtime");
+
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteSegmentWithGetAPI(TABLE_NAME, realtimeSegmentsList.get(0).toString(), "realtime"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    JSONArray realtimeSegmentsListReturn = getSegmentsFromJson(postDeleteSegmentList, "realtime");
+    realtimeSegmentsList.remove(0);
+    Assert.assertEquals(realtimeSegmentsListReturn, realtimeSegmentsList);
+  }
+
+  // @Test Commenting this out for now because of HLC/LLC
+  public void deleteAllRealtimeSegmentsFromDeleteAPI() throws Exception{
+    sendDeleteRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forSegmentDeleteAllAPI(TABLE_NAME, "realtime"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 1);
+    Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "offline").size(), 0);
+
+     repushRealtimeSegments();
+  }
+
+    @Test
+    public void deleteAllOfflineSegmentsFromDeleteAPI() throws Exception{
+      sendDeleteRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+          forSegmentDeleteAllAPI(TABLE_NAME, "offline"));
+      Thread.sleep(3000);
+
+      String postDeleteSegmentList = sendGetRequest(
+          ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+      Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "offline"), null);
+      Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 0);
+
+      repushOfflineSegments();
+    }
+
+    @Test
+    public void deleteOfflineSegmentFromDeleteAPI() throws Exception{
+      String segmentList = sendGetRequest(
+          ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+      JSONArray offlineSegmentsList = getSegmentsFromJson(segmentList, "offline");
+
+      sendDeleteRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+          forSegmentDeleteAPI(TABLE_NAME, offlineSegmentsList.get(0).toString(), "offline"));
+      Thread.sleep(3000);
```

Comment:
remove sleep add a loop, and also remember segment name and look for it in the segment list